core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -121,7 +120,7 @@ public void update(String memberId, FetchPartitionData fetchPartitionData) {
         }
     }
 
-    public CompletableFuture<Optional<Throwable>> acknowledge(String memberId, List<AcknowledgementBatch> acknowledgementBatch) {
+    public CompletableFuture<Errors> acknowledge(String memberId, List<AcknowledgementBatch> acknowledgementBatch) {
```

Comment:
Any reason for this change?

Reply:
We are using Errors everywhere, so I thought returning an object of Errors class would make more sense.